lib/baseFactory.js

```diff
@@ -171,7 +176,8 @@ class BaseFactory {
                 throw new Error(`No datastore provided to ${className}`);
             }
 
-            if (!config || !config.scmPlugin) {
+            // SecretFactory does not use ScmPlugin
+            if (className !== 'SecretFactory' && (!config || !config.scmPlugin)) {
```

Comment:
I don't like the idea of tying baseFactory to a specific factory implementation. Maybe this check should get moved into the specific factory instances that actually need it. (Job doesn't actually need it either.)